### PR TITLE
Fixes #32990 - job wizard select default template

### PR DESCRIPTION
**Pull request by MariaAga**
The default template is set in the Remote Execution settings
depends on: https://github.com/theforeman/foreman_remote_execution/pull/629

### CODE REVIEW

File: app/controllers/ui_job_wizard_controller.rb [hunk 1]
```diff
@@ -24,6 +24,19 @@ def map_template_inputs(template_inputs_with_foreign)
     template_inputs_with_foreign.map { |input| input.attributes.merge({:resource_type_tableize => input.resource_type&.tableize }) }
   end
 
+  def default_category
+    default_template&.job_category
+  end
+
+  def default_template
```
Comment (adamruzicka) [inline app/controllers/ui_job_wizard_controller.rb:31]: This is almost the same as https://github.com/theforeman/foreman_remote_execution/blob/master/app/models/job_invocation_composer.rb#L643 . For now I wouldn't mind having it duplicated and I'd like to see it eventually removed from the composer once we fully move to the new form.

However, the implementation in composer currently checks permissions, which this one does not. We should check them here as well

**Review by adamruzicka — CHANGES_REQUESTED** [review]
One change needed, apart from it it works nicely

File: app/controllers/ui_job_wizard_controller.rb [hunk 2]
```diff
@@ -24,6 +24,19 @@ def map_template_inputs(template_inputs_with_foreign)
...
+    setting_value = Setting['remote_execution_form_job_template']
+    return nil unless setting_value
```
Comment (adamruzicka) [inline app/controllers/ui_job_wizard_controller.rb:33]: `return nil` is equivalent to just `return`, there's no need to be this explicit

File: app/controllers/ui_job_wizard_controller.rb [hunk 3]
```diff
@@ -24,6 +24,19 @@ def map_template_inputs(template_inputs_with_foreign)
...
+
+    template = JobTemplate.authorized(:view_job_templates).find_by :name => setting_value
+    return nil unless template
+    template
```
Comment (adamruzicka) [inline app/controllers/ui_job_wizard_controller.rb:37]: `find_by` returns either the thing or nil. We don't really have to check if it is nil to return nil or the thing itself

File: app/controllers/ui_job_wizard_controller.rb [hunk 4]
```diff
@@ -24,6 +24,19 @@ def map_template_inputs(template_inputs_with_foreign)
...
+  end
```
Comment (adamruzicka) [inline app/controllers/ui_job_wizard_controller.rb:38]: All in all we could even do something like this, but that boils down to personal preference. Either way is fine as long we don't do unneeded checks

```suggestion
  def default_template
    if setting_value = Setting['remote_execution_form_job_template']
      JobTemplate.authorized(:view_job_templates).find_by :name => setting_value
    end
  end
```

Comment (MariaAga) [reply app/controllers/ui_job_wizard_controller.rb]: Used this instead, with added parenthesis to make rubocop happy.
Thanks for the review!

**Review by adamruzicka — APPROVED** [review]
Works nicely

**Comment by adamruzicka** [comment]
Thank you @MariaAga !

**Pull request merged**